src/slam_toolbox_common.cpp

```diff
@@ -300,12 +300,13 @@ bool SlamToolbox::shouldStartWithPoseGraph(
 {
   // if given a map to load at run time, do it.
   this->declare_parameter("map_file_name", std::string(""));
-  this->declare_parameter("map_start_pose", std::vector<double>());
-  this->declare_parameter("map_start_at_dock", false);
+  auto read_pose_param = this->declare_parameter("map_start_pose");
```

Comment:
Please rename these 2 params to `map_start_pose` and `start_at_doc`. Adding a `_param` is redundant. All objects are basically params so its not really meaningful